frontend/src/components/ride/RideProfile.tsx

```diff
@@ -0,0 +1,112 @@
+// TODO: recreate the userform and modify for ride.
+// Make sure to look into the context and see if it is needed.
+
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import { Button, Form } from 'reactstrap';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import RideForm from '../forms/RideForm';
+
+// type RideType = { road: boolean; mountain: boolean; other: boolean };
+// type RideDifficulty = { hard: boolean; medium: boolean; easy: boolean };
+
+export interface FormFields {
+  rideTitle: string;
+  rideDescription: string;
+  rideType: string;
+  rideDifficulty: string;
+}
+
+export interface RideFormProps {
+  readOnly?: boolean;
+}
+
+function RideProfile(props: { location: string; createRide: boolean }) {
+  const [ride, setRide] = useState({});
+  const [isValidating, setisValidating] = useState([]);
+  const [readOnly, setReadOnly] = useState(!props.createRide);
+
+  const navigate = useNavigate();
+
+  // TODO: the following is just an experiment to test the db. modify when done.
+  useEffect(() => {
+    const rideId = '6504c7fc0afe1183e4538209';
+    axios.get(`/api/rides/${rideId}`).then((res) => {
+      console.log('ride: ', res.data);
+      setRide(res.data);
+    });
+  }, []);
+
+  const useFormReturn = useForm<FormFields>({
+    mode: 'all',
+    reValidateMode: 'onChange',
+    defaultValues: {
+      rideTitle: 'Short ride',
+      rideDescription: 'This is a somewhat longer description of the ride',
+    },
+  });
+
+  const onFormSubmit = (data: any) => {
+    // TODO: don't submit on page load/refresh
+    // e.preventDefault();
+    // await data;
+    console.log('**************** data: ', data);
+    // console.log('**************** in create ride', data);
+    // axios
+    //   .post('/api/ride/create')
+    //   .then(res => {
+    //     setRide(res.data);
+    //     console.log(res.data);
+    //   }
+    // );
+    reset(data);
+    setReadOnly(true);
+    // navigate('/');
+    // TODO: validate radio buttons on submit: at least one per type
+    // needs to be checked.
```

Comment:
we need to clean out all comments in all files, we can create a ticket for that.